doc/user-manual/getting-started/hello-world.lagda.rst

```diff
@@ -7,35 +7,453 @@
+``?0 : x + (y + z) ≡ x + y + z``, which is the proof of ``+assoc x y z``.
+
+As it is structurally smaller than ``+assoc (suc x) y z``, we can recursively
+use it as a proof. Agda performs
+:ref:`termination checking <termination-checking>` on recursive functions.
+Note that not all recusions are allowed, only the ones that are mechanically
+proved to terminate, like in this case.
+
+The result of the definition we were looking for is:
+
+.. code-block:: agda
+
+  +assoc x y z = refl
+  +assoc (suc x) y z = cong suc (+assoc x y z)
+
+Now just load the file again and you will see ***All Done***. This means that
+indeed ``+assoc`` is a member of ``+assoc-enun`` and therefore its proof.
+
+.. important::
+  The ``x`` in the type signature of ``+assoc`` is **not** the same as the
+  ``x`` pattern variable in the last clause where ``suc x`` is written. The
+  following would work also: ``+assoc (suc x₁) y z = cong suc (+assoc x₁ y z)``.
+  The scope of a variable declared in a signature is restricted to the
+  signature itself.
+
+Here is the final code of the ‘Hello world’ proof example:
+
+.. code-block:: agda
+
+  module hello-world-proof where
+
+  open import Agda.Builtin.Nat using (Nat ; _+_)
+  open import Agda.Builtin.Equality using (_≡_)
+
+  +assoc-enun : Set
+  +assoc-enun = ∀ (x y z : Nat) → x + (y + z) ≡ (x + y) + z
+
+  open import Agda.Builtin.Nat using (zero ; suc)
+  open import Relation.Binary.PropositionalEquality using (cong)
+  open import Agda.Builtin.Equality using (refl)
+
+  +assoc : ∀ (x y z : Nat) → x + (y + z) ≡ (x + y) + z
+  +assoc zero y z = refl
+  +assoc (suc x) y z = cong suc (+assoc x y z)
+
+.. note:: You can learn more details about proving in the chapter
+  `Proof by Induction <plfa-induction_>`_ of the
+  `Programming Language Foundations in Agda <plfa_>`_ online book.
+
+.. _plfa-induction: https://plfa.github.io/Induction/
+.. _plfa: https://plfa.github.io
+
+‘Hello world’ program
+=====================
+
+Agda is a dependently typed functional programming language. This entails the
+fact that it is possible to write programs in Agda that interact with the
+world. In this section, we will write a first ‘Hello world’ program in Agda
+that we will be able to compile and execute right away.
+
+Agda Source Code
+----------------
+
+First, we create a new file named ``hello-world-prog.agda`` with Emacs or Atom
+in a folder that we will refer to as our top-level folder.
+
+.. code-block:: agda
+
+  module hello-world-prog where
+
+  open import Agda.Builtin.IO using (IO)
+  open import Agda.Builtin.Unit using (⊤)
+  open import Agda.Builtin.String using (String)
+
+  postulate putStrLn : String → IO ⊤
+  {-# FOREIGN GHC import qualified Data.Text as T #-}
+  {-# COMPILE GHC putStrLn = putStrLn . T.unpack #-}
+
+  main : IO ⊤
+  main = putStrLn "Hello world!"
+
+This code is self-contained and has several declarations:
+
+1. imports of the ``ÌO``, ``⊤`` and ``String`` datatypes from the
```

Comment:
"Standard library" usually refers to the, er, standard library at https://github.com/agda/agda-stdlib.